src/com/dabomstew/pkrandom/romhandlers/AbstractRomHandler.java

```diff
@@ -1930,6 +1930,9 @@ public void randomizeTrainerHeldItems(Settings settings) {
         boolean giveToImportantPokemon = settings.isRandomizeHeldItemsForBossTrainerPokemon();
         boolean giveToRegularPokemon = settings.isRandomizeHeldItemsForBossTrainerPokemon();
         boolean highestLevelOnly = settings.isHighestLevelGetsItemsForTrainers();
+
+        List<Move> moves = this.getMoves();
+        Map<Integer, List<MoveLearnt>> movesets = this.getMovesLearnt();

```

Comment:
I have it here because I observed noticeable added latency when running locally and attributed it to fetching the moves and movesets repeatedly. Calling these two functions takes a few milliseconds and since it's static data, it didn't make sense to me to call it inside a double for loop. Rough numbers: 3 ms * 600 trainers * 2 pokes / trainer = 3 extra seconds of latency.

To me, that seems like an easy win to keep the moves/movesets here. Notice that we don't throw them away until we've randomized all the trainers and this function only gets called when at least some trainers get randomized.